### ISSUE TITLE
WPF: HasNavigationBar doesn't have any effect

### ISSUE THREAD

**Issue by debugnik** [fsprojects/Fabulous#223]
I'm using Fabulous.Core 0.25.0 with Xamarin.Forms 3.3.0.967583 on WPF. To be honest, I can't figure out if this is a bug in Fabulous or in Xamarin's WPF platform.

When trying to create an empty view without navigation bar

```f#
View.ContentPage().HasNavigationBar(false)
```

`HasNavigationBar` doesn't seem to have any effect; the navigation bar is still there. Note that

```f#
View.ContentPage(created=fun page -> NavigationPage.SetHasNavigationBar(page, false))
```

does actually hide the navigation bar. I'm assuming that the former already works in other platforms.

**Comment by TimLariviere** [comment]
I can't reproduce.
Tried both on Android, iOS and WPF with the given versions and they all reacted properly (hiding NavBar if false, showing if true or not specified).

Here's what I tried
```fsharp
let view (model: Model) dispatch =
    View.NavigationPage(
        pages=[
            View.ContentPage(title="Title").HasNavigationBar(false)
        ]
    )
```
Note: Tried also to set/remove the title property, still working as expected.

Do you have or could you make a reproduction of your issue?
Thanks

**Comment by TimLariviere** [comment]
Oh ok. I have the same issue with only `View.ContentPage().HasNavigationBar(false)`
Seems to be specific to Xamarin.Forms.Platform.WPF.

Having a navigation bar without a NavigationPage doesn't really make much sense.
Android and iOS just don't show at all the page's title if there's no NavigationPage, only WPF does.

Maybe open an issue on https://github.com/xamarin/Xamarin.Forms?

**Comment by mistiara** [comment]
> Having a navigation bar without a NavigationPage doesn't really make much sense.

Yeah, but ContentPage comes with an (empty) toolbar by default and that is apparently considered part of a navigation bar, so it creates a big empty space.

I'll take a look at how this is expected to work on Xamarin.Forms and will consider opening an issue there. Thanks for helping me figure this out!

**Comment by johannesegger** [comment]
@debugnik did you find anything? I don't understand why the navigation bar shows up at all.

**Comment by TimLariviere** [comment]
@debugnik @johannesegger Oh I think I know why and this is certainly an issue in Fabulous.

`HasNavigationBar` is an attached property from `NavigationPage`.
Usually those attached properties are attached to direct children and processed by the parent (ie ContentPages inside a NavigationPage).
Fabulous follows this convention, and only applies attached property values from the parent level.
So if you don't have a parent `NavigationPage`, `HasNavigationBar` won't be applied to the `ContentPage`.

But in this specific case, the attached property seems to have an effect even without the parent NavigationPage.

https://github.com/fsprojects/Fabulous/blob/dcf5dbe175fa617977a6aa08a5a25dff25f5c292/tools/Generator/Xamarin.Forms.Core.json#L1707-L1731

The way the Generator handles attached properties is currently leading to several issues.
Really need to rethink this in the future.

For now, one way to fix that would be to do like this:
https://github.com/fsprojects/Fabulous/blob/dcf5dbe175fa617977a6aa08a5a25dff25f5c292/tools/Generator/Xamarin.Forms.Core.json#L43-L47

`ShellBackgroundColor` is an attached property from Shell, but can be applied to any level of children.
This is not supported by the Generator, so I had to add this attached property to `Element` as a fictitious property. `updateShellBackgroundColor` calls `Shell.SetBackgroundColor`.

**Comment by debugnik** [comment]
Exactly, all pages seem to have a NavigationBar despite not being navigation pages, because the same bar is used for toolbars. In XAML, this is controlled by the `NavigationPage.HasNavigationBar` attribute but fabulous only applied it to NavigationPage.

Sorry I didn't follow up on this, I had to put my fabulous project aside for a while.

**Comment by johannesegger** [comment]
Ok that sounds plausible, thanks for the explanation. Though Xamarin.Forms' behavior that the navigation bar is visible in WPF apps only seems to be inconsistent as well.

**Comment by johannesegger** [comment]
Ohh, because in my case the property never changes (I always want the navigation bar to be invisible) it was easier than I thought (thanks for the hint what property I have to set):

```fsharp
View.ContentPage(
    created = (fun e -> NavigationPage.SetHasNavigationBar(e, false))
)
```

**Comment by TimLariviere** [comment]
It is fixed in v0.50.0-alpha.2

**Comment by TimLariviere** [comment]
This will be released in v0.50.0 which will arrive in the coming days (once we add support for XF 4.3)
